include/librealsense2/hpp/rs_options.hpp

```diff
@@ -116,11 +128,23 @@ namespace rs2
             return res > 0;
         }
 
+        /**
+         * sets a callback in case an option in this options container value is updated
+         * \param[in] callback     the callback function
+         */
+        template< class T >
+        void on_options_value_update( T callback ) const
```

Comment:
This should be `on_options_changed`, to fit the rs2_ function name

Reply:
Also, can you please try to change this to just `std::function< void( ... ) >`? I really don't think a template is needed here.

Reply:
Renamed